runtime/j9vm/jvm.c

```diff
@@ -2135,7 +2137,7 @@ jint JNICALL JNI_CreateJavaVM(JavaVM **pvm, void **penv, void *vm_args) {
 						&xServiceBuffer, argEncoding, localVerboseLevel))
 #ifdef J9VM_OPT_HARMONY
 				/* pass in the Harmony library */
-				|| (0 != addHarmonyPortLibrary(&j9portLibrary, &vmArgumentsList, localVerboseLevel))
+				|| (((j2seVersion & J2SE_SERVICE_RELEASE_MASK) < J2SE_19) && (0 != addHarmonyPortLibrary(&j9portLibrary, &vmArgumentsList, localVerboseLevel)))
```

Comment:
Only IBM builds of Java 8 need the harmony port library. OpenJ9 Java 8 builds do not.

Reply:
Peter
Thanks. What's the runtime/build flag for IBM vs Harmony?

Reply:
We could try to re-use J9VM_OPT_HARMONY, disable this flag in OpenJ9, and then enable it internally to IBM, although I suspect uma would get in the way. It should be easier when cmake becomes available. Alternatively a new flag can be added (OPENJ9_OPT_HARMONY?), outside of uma control and then it should be fairly straight forward. All (3) uses of J9VM_OPT_HARMONY could be replaced by OPENJ9_OPT_HARMONY, and then the following will keep J9VM_OPT_HARMONY working.

```
#if defined(J9VM_OPT_HARMONY)
#define OPENJ9_OPT_HARMONY
#endif
```

Reply:
We'd want something like the following in the module.xml's for the files affected. Then its just a matter of fixing the IBM builds to enable OPENJ9_OPT_HARMONY, probably using sed to modify mkconstants.mk before building.

```
		<makefilestubs>
			<makefilestub data="ifeq ($(OPENJ9_OPT_HARMONY),true)"/>
			<makefilestub data="vmifunc$(UMA_DOT_O) : CFLAGS += -DOPENJ9_OPT_HARMONY"/>
			<makefilestub data="endif"/>
		</makefilestubs>
```

Reply:
Yeah, I was going to try disabling J9VM_OPT_HARMONY  in the Panama spec, since it's Java 9 only.  I was planning to do the serious build flag work when we switch to cmake.

In the meantime, can we merge this (incremental) piece?

Thanks

Reply:
I don't see any reason to deliver it as it doesn't improve the build / porting story. We'll still need the harmony library for OpenJ9 Java 8 and any benefit we were hoping to achieve won't be realized.

Reply:
Alternatively the `-D/-U` options I added to UMA could come in handy here. It lets you override buildspec flag settings on the command line. This is done by passing `-D <opt>` or `-U <opt>` to set/unset the option respectively.  ie `-D opt_harmony`